app/models/branch.rb

```diff
@@ -13,21 +13,23 @@ class Branch < ApplicationRecord
   # callbacks
   after_create :set_inventories
 
+
   def available_meals
-    available_meals = []
-    Meal.includes(:meal_items).includes(:ratings).all.each { |meal| available_meals << meal if sufficient_stock?(meal) }
-    available_meals
+    Meal.includes(:meal_items).includes(:ratings).select { |meal| meal if sufficient_stock?(meal) && meal.active }
```

Comment:
I don't think this will work. Please verify.

Reply:
discussed